packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,388 @@
+  statusMap,
+  errorPopup,
+  processingPopup,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <GalleryItem className="inventory-card-item">
+      {listLoaded && !listError ? (
+        <>
+          <div className="inventory-card-sub-item">
+            <Link to={redirectPath}>
+              {t('{{count}} {{resourceType}}', {
+                count: listItems.length,
+                resourceType: resource,
+              })}
+            </Link>
+          </div>
+          {statusMap && statusMap[PhaseType.ERROR] && (
+            <div className="icons-container">
+              <RedExclamationCircleIcon
+                title={PhaseType.ERROR}
+                className="icons"
+              />
+              {errorPopup}
+            </div>
+          )}
+          {statusMap && statusMap[PhaseType.PROCESSING] && (
+            <div>
+              <BlueInProgressIcon
+                title={PhaseType.PROCESSING}
+                className="icons"
+              />
+              {processingPopup}
+            </div>
+          )}
+        </>
+      ) : (
+        <Skeleton
+          screenreaderText={t('Loading {{resourceType}}', {
+            resourceType: resource,
+          })}
+        />
+      )}
+    </GalleryItem>
+  );
+};
+
+const BucketPopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <BucketClassPopOver
+      label={String(statusmap[type])}
+      bucketClasses={data as string[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('Buckets: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
+
+const DataResourcePopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <DataResourcesPopOver
+      label={String(statusmap[type])}
+      dataResources={data as string[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('Data sources: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
+
+const OBCPopup: React.FC<PopupProps> = ({ type, statusmap, data }) => {
+  const { t } = useTranslation();
+  return (
+    <OBCPopOver
+      label={String(statusmap[type])}
+      obcDetails={data as obcType[]}
+      headerContent={getHeaderHTMLElement(
+        type,
+        t('ObjectBucketClaims: {{type}}', {
+          type: type,
+        })
+      )}
+      trimContent
+    />
+  );
+};
+
+export const InventoryCard: React.FC = () => {
+  const { t } = useTranslation();
+  const [buckets, bucketsLoaded, bucketsError] =
+    useK8sWatchResource<BucketClassKind[]>(bucketClassResource);
+  const [dataResources, dataResourcesLoaded, dataResourcesError] =
+    useK8sWatchResource<NamespaceStoreKind[]>(nameSpaceStoreResource);
+  const [obc, obcLoaded, obcError] =
+    useK8sWatchResource<K8sResourceKind[]>(bucketClaimResource);
+
+  const {
+    statusMap: bucketStatusMap,
```

Comment:
why are we using this `bucketStatusMap` variable at all ??
we have `errorList` and we have `processingList` we can directly do `errorList?.length` and `processingList?.length` why are we storing count once again in `bucketStatusMap`.

Reply:
we need this to keep track of all the errors and processing of all 3 types of resources in one place.

if we are removing this, then there is a problem as each popup has it's own key for the list 

![image](https://user-images.githubusercontent.com/22158580/173068868-47f1c922-d2e4-430a-857e-bcc0f732d809.png)

Reply:
children?.[0] can be bucketClasses / dataResource / obcList.

so we can't have resource-specific things in the generic component and need to pass something which has all three info

Reply:
done created map like this :+1: 

![image](https://user-images.githubusercontent.com/22158580/173091926-5bc7932c-6080-42fa-a7c8-90c2cd005f76.png)

Reply:
looks good !
one last thing, do we need to store `Ready` ones as well ? because I think we don't need that right ?

Reply:
removed ready state 

![image](https://user-images.githubusercontent.com/22158580/173302899-c0d33c14-ffbf-40f4-b805-69c1719ffa5b.png)